modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/OrganizationODataRetrieverCustomFieldsTest.java

```diff
@@ -334,6 +550,150 @@ public void testGetOrganizationsFilterByCustomFieldWithEqualsAndLocalizedStringT
 		Assert.assertEquals(0, count);
 	}
 
+	@Test
+	public void testGetOrganizationsFilterByCustomFieldWithEqualsAndLongKeywordType()
+		throws Exception {
+
+		ExpandoColumn expandoColumn = _addExpandoColumn(
+			_expandoTable, RandomTestUtil.randomString(),
+			ExpandoColumnConstants.LONG,
+			ExpandoColumnConstants.INDEX_TYPE_KEYWORD);
+
+		long columnValue = 3;
+
+		Organization organization1 = _addOrganization(
+			expandoColumn.getName(), columnValue);
+
+		_organizations.add(organization1);
+
+		Organization organization2 = OrganizationTestUtil.addOrganization();
+
+		_organizations.add(organization2);
+
+		String filterString = String.format(
+			"(customField/%s eq %s)", _encodeName(expandoColumn), columnValue);
+
+		int count = _getODataRetriever().getResultsCount(
+			TestPropsValues.getCompanyId(), filterString,
+			LocaleUtil.getDefault());
+
+		Assert.assertEquals(1, count);
+
+		List<Organization> organizations = _getODataRetriever().getResults(
+			TestPropsValues.getCompanyId(), filterString,
+			LocaleUtil.getDefault(), 0, 1);
+
+		Assert.assertEquals(organization1, organizations.get(0));
+	}
+
+	@Test
+	public void testGetOrganizationsFilterByCustomFieldWithEqualsAndLongTextType()
+		throws Exception {
+
+		ExpandoColumn expandoColumn = _addExpandoColumn(
+			_expandoTable, RandomTestUtil.randomString(),
+			ExpandoColumnConstants.LONG,
+			ExpandoColumnConstants.INDEX_TYPE_TEXT);
+
+		long columnValue = 3;
+
+		Organization organization1 = _addOrganization(
+			expandoColumn.getName(), columnValue);
+
+		_organizations.add(organization1);
+
+		Organization organization2 = OrganizationTestUtil.addOrganization();
+
+		_organizations.add(organization2);
+
+		String filterString = String.format(
+			"(customField/%s eq %s)", _encodeName(expandoColumn), columnValue);
+
+		int count = _getODataRetriever().getResultsCount(
```

Comment:
Ohhh, sorry, chaining is not allowed here .

Reply:
Since `_getODataRetriever()`  is a method class (without using `this`), chaining is allowed here, actually source formatter didn't complain about it. For me It's ok like it is, what do you think?

Reply:
I know SF don't complain, since it is green in the pull, but I think Brian Chan won't allow this method

Reply:
But I agree with you, if SF is not complaining, better if we don't change it, and see is Brian has an opinion about it.